extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationSearch.js

```diff
@@ -0,0 +1,36 @@
+$(function ($) {
+	'use strict';
+
+	var $globalNav = $('.wds-global-navigation'),
+		$searchInput = $globalNav.find('.wds-global-navigation__search-input'),
+		$searchSubmit = $globalNav.find('.wds-global-navigation__search-submit'),
+		placeHolderText = $searchInput.attr('placeholder'),
+		activeSearchClass = 'wds-search-is-active';
+
+	$searchInput.on('focus', function () {
+		if (!$globalNav.hasClass(activeSearchClass)) {
+			$globalNav.addClass(activeSearchClass);
+			$searchInput.attr('placeholder', $searchInput.data('active-text'));
+		}
+	});
+
+	// Make sure default state is correct when navigating back in Chrome
+	if ($searchInput.val().length > 0 && $searchSubmit.prop('disabled')) {
+		$searchSubmit.prop('disabled', false);
+	}
+
+	$searchInput.on('input', function () {
+		var textLength = $(this).val().length;
+
+		if (textLength > 0 && $searchSubmit.prop('disabled')) {
+			$searchSubmit.prop('disabled', false);
+		} else if (textLength === 0) {
```

Comment:
Sorry, my bad. `else` is not enough. Button would get enabled/disabled on every letter.